pandas/core/generic.py

```diff
@@ -8825,22 +8829,39 @@ def _where(
             if other.ndim <= self.ndim:
 
                 _, other = self.align(
-                    other, join="left", axis=axis, level=level, fill_value=np.nan
+                    other,
+                    join="left",
+                    axis=axis,
+                    level=level,
+                    fill_value=np.nan,
+                    copy=False,
                 )
 
                 # if we are NOT aligned, raise as we cannot where index
-                if axis is None and not all(
-                    other._get_axis(i).equals(ax) for i, ax in enumerate(self.axes)
-                ):
+                if axis is None and not other._indexed_same(self):
                     raise InvalidIndexError
 
+                elif other.ndim < self.ndim:
+                    # TODO(EA2D): avoid object-dtype cast in EA case GH#38729
+                    other = other._values
+                    if axis == 0:
+                        other = np.reshape(other, (-1, 1))
+                    if axis == 1:
+                        other = np.reshape(other, (1, -1))
+
+                    other = np.broadcast_to(other, self.shape)
```

Comment:
When does this happen? When `other` is a Series, I think? 
In that case you can also broadcast the Series to an aligned DataFrame without the conversion to numpy?

Reply:
Also, does this fix a case? Does it need a test?

Reply:
> When other is a Series, I think?

Yes.

> In that case you can also broadcast the Series to an aligned DataFrame without the conversion to numpy?

Most ways of doing that would make copies on DataFrame construction, which this avoids.

> Also, does this fix a case? Does it need a test?

Yes, fixes #38729, will add tests.